packages/google_maps_flutter/google_maps_flutter/example/integration_test/src/tiles_inspector.dart

```diff
@@ -208,269 +207,574 @@ void runTests() {
-      WeightedLatLng(LatLng(37.785, -122.445)),
-      WeightedLatLng(LatLng(37.785, -122.443)),
-      WeightedLatLng(LatLng(37.785, -122.441)),
-      WeightedLatLng(LatLng(37.785, -122.439)),
-      WeightedLatLng(LatLng(37.785, -122.437)),
-      WeightedLatLng(LatLng(37.785, -122.435), weight: 2)
-    ],
-    dissipating: false,
-    gradient: HeatmapGradient(
-      <HeatmapGradientColor>[
-        HeatmapGradientColor(
-          Color.fromARGB(255, 0, 255, 255),
-          0.2,
+    testWidgets('set heatmap correctly', (WidgetTester tester) async {
+      final Completer<int> mapIdCompleter = Completer<int>();
+      final Heatmap heatmap2 = Heatmap(
+        heatmapId: const HeatmapId('heatmap_2'),
+        data: heatmap1.data,
+        dissipating: heatmap1.dissipating,
+        gradient: heatmap1.gradient,
+        maxIntensity: heatmap1.maxIntensity,
+        opacity: heatmap1.opacity - 0.1,
+        radius: heatmap1.radius,
+        minimumZoomIntensity: heatmap1.minimumZoomIntensity,
+        maximumZoomIntensity: heatmap1.maximumZoomIntensity,
+      );
+
+      await tester.pumpWidget(
+        Directionality(
+          textDirection: TextDirection.ltr,
+          child: GoogleMap(
+            initialCameraPosition: kInitialCameraPosition,
+            heatmaps: <Heatmap>{heatmap1, heatmap2},
+            onMapCreated: (GoogleMapController controller) {
+              mapIdCompleter.complete(controller.mapId);
+            },
+          ),
         ),
-        HeatmapGradientColor(
-          Color.fromARGB(255, 0, 63, 255),
-          0.4,
+      );
+      await tester.pumpAndSettle(const Duration(seconds: 3));
+
+      final int mapId = await mapIdCompleter.future;
+      final GoogleMapsInspectorPlatform inspector =
+          GoogleMapsInspectorPlatform.instance!;
+
+      if (inspector.supportsGettingHeatmapInfo()) {
+        final Heatmap heatmapInfo1 =
+            (await inspector.getHeatmapInfo(heatmap1.mapsId, mapId: mapId))!;
+        final Heatmap heatmapInfo2 =
+            (await inspector.getHeatmapInfo(heatmap2.mapsId, mapId: mapId))!;
+
+        expectHeatmapEquals(heatmap1, heatmapInfo1);
+        expectHeatmapEquals(heatmap2, heatmapInfo2);
+      }
+    });
+
+    testWidgets('update heatmaps correctly', (WidgetTester tester) async {
+      final Completer<int> mapIdCompleter = Completer<int>();
+      final Key key = GlobalKey();
+
+      await tester.pumpWidget(
+        Directionality(
+          textDirection: TextDirection.ltr,
+          child: GoogleMap(
+            key: key,
+            initialCameraPosition: kInitialCameraPosition,
+            heatmaps: <Heatmap>{heatmap1},
+            onMapCreated: (GoogleMapController controller) {
+              mapIdCompleter.complete(controller.mapId);
+            },
+          ),
         ),
-        HeatmapGradientColor(
-          Color.fromARGB(255, 0, 0, 191),
-          0.6,
+      );
+      await tester.pumpAndSettle(const Duration(seconds: 3));
```

Comment:
Without this line tests timeouted on Web platform, and we were not able to test ground overlay below.

Reply:
Were they timing out in all tests, or wasm specifically? I know I had to revisit a bunch of plugins and add a few `pump`s and `pumpAndSettle` when enabling the wasm tests. See:

* https://github.com/flutter/packages/pull/8111